tests/Parsers/Utils.php

```diff
@@ -24,4 +30,52 @@ public static function tagDoesNotHaveZeroPatchVersion(Since|RemovedTag|Deprecate
     {
         return (bool)preg_match('/^[1-9]+\.\d+(\.[1-9]+\d*)*$/', $tag->getVersion()); //find version like any but 7.4.0
     }
+
+    public static function getDeclaredSinceVersion(BasePHPElement $element): ?string
+    {
+        $allSinceVersions = [];
+        if ($element instanceof PHPMethod) {
+            $parentClass = PhpStormStubsSingleton::getPhpStormStubs()->getClass($element->parentName);
+            if ($element->hasInheritDocTag || $element->parentName === '___PHPSTORM_HELPERS\object') {
+                return null;
+            }
+            if (!empty($element->sinceTags)) {
+                $allSinceVersions[] = array_map(fn(Since $tag) => $tag->getVersion(), $element->sinceTags);
+            }
+            if ($element->sinceVersionFromAttribute !== null) {
+                $allSinceVersions[] = $element->sinceVersionFromAttribute;
+            }
+            if (!empty($parentClass->sinceTags)) {
+                $allSinceVersions[] = array_map(fn(Since $tag) => $tag->getVersion(), $parentClass->sinceTags);
+            }
+        } elseif ($element instanceof PHPFunction) {
+            if (!empty($element->sinceTags)) {
+                $allSinceVersions[] = array_map(fn(Since $tag) => $tag->getVersion(), $element->sinceTags);
+            }
+            if ($element->sinceVersionFromAttribute !== null) {
+                $allSinceVersions[] = $element->sinceVersionFromAttribute;
+            }
+        } elseif ($element instanceof PHPDefineConstant) {
```

Comment:
The duplicated part:
```
if (!empty($element->sinceTags)) {
                $allSinceVersions[] = array_map(fn(Since $tag) => $tag->getVersion(), $element->sinceTags);
            }
            if ($element->sinceVersionFromAttribute !== null) {
                $allSinceVersions[] = $element->sinceVersionFromAttribute;
            }
```
can be extracted out of if statement.

Reply:
Very valid point! Thank you! fixed that. 👍